src/engine/execute.js

```diff
@@ -198,6 +227,9 @@ var execute = function (sequencer, thread) {
         getParam: function (paramName) {
             return thread.getParam(paramName);
         },
+        setProcedureReturn: function (value) {
+            thread.stackFrames[0].executionContext.return = value;
```

Comment:
Wouldn't that cause the value to be set to the report block instead of the top block (hat)